src/main/kotlin/com/dprint/services/editorservice/EditorProcess.kt

```diff
@@ -147,14 +149,30 @@ class EditorProcess(private val project: Project) {
     fun readInt(): Int {
         val stdout = getProcess().inputStream
         val result = ByteBuffer.wrap(stdout.readNBytes(U32_BYTE_SIZE)).int
-        LOGGER.debug(Bundle.message("formatting.received.value", result))
+        LOGGER.info(Bundle.message("formatting.received.value", result))
         return result
     }
 
     fun readString(): String {
-        val stdout = getProcess().inputStream
         val totalBytes = readInt()
+        val result = readBuffer(totalBytes)
+        val decodedResult = result.decodeToString()
+        LOGGER.info(Bundle.message("formatting.received.value", decodedResult))
+        return decodedResult
+    }
+
+    fun writeBuffer(byteArray: ByteArray) {
+        val stdin = getProcess().outputStream
+        stdin.write(byteArray)
+        stdin.flush()
+    }
+
+    fun readBuffer(totalBytes: Int): ByteArray {
+        val stdout = getProcess().inputStream
         var result = ByteArray(0)
```

Comment:
In kotlin, is it possible to set the `totalBytes` here in order to pre-allocate the array ahead of time? Might be a bit of a performance improvement.

Reply:
Yeah it is it's just a bit of a pain to allocate them as you need to add them one by one I think.